app/content-panel/blogs/[id]/edit/page.tsx

```diff
@@ -0,0 +1,187 @@
+'use client';
+
+import React, { useState, useEffect, useRef } from 'react';
+import { useRouter } from 'next/navigation';
+import styles from '../BlogDetailsPage.module.scss';
+import BlogDetailsCard from '../../../../../components/reusable-components/blogDetails-card/BlogDetailsCard';
+
+interface Author {
+  first_name: string;
+  last_name: string;
+}
+
+interface BlogDetails {
+  title: string;
+  content: string;
+  image: string;
+  author: Author;
+  publishDate: string;
+  tags: string[];
```

Comment:
We dont need to create another page to be able to edit the blog post. It will remain the same page. On Click on Edit all inputs will become enabled and can be edited.

Reply:
Additionally this should be on another branch after we merge this one, which is made for the ticket: view individual blog page.
You can revert this changes with git. Look into how you can see the last stable commit(the one before these changes with edit), and revert the branch head to that commit :D
Let me know if you need any help doing that.

Reply:
> We dont need to create another page to be able to edit the blog post. It will remain the same page. On Click on Edit all inputs will become enabled and can be edited.

My previous code met these requirements, but I considered creating a separate edit page to enable redirection from the Blog List page when clicking the edit button. Without a separate page, I'm unsure how to access the edit mode directly.
![Screenshot 2024-08-29 231108](https://github.com/user-attachments/assets/02b175eb-06c7-49d7-a00a-405e10d3cbdd)



What do you think, should I bring back my old code, or its ok to stay like it is now ?

Reply:
Or should I revert the changes and move the current code to a new branch called edit-blog, or is the separate EditBlog page unnecessary altogether?

Reply:
The problem here is:
You have `blogs page`and `individual blog page`. On Click on **Edit** on the `blogs page` you should be redirected to the `individual blog page`, but the `individual blog page` should be in **_edit mode_**.
How can these two pages have that state available to both of them?

Reply:
The EditBlog is unnecessary. @mitkapanarin